lib/src/fty_shm.cc

```diff
@@ -455,58 +405,84 @@ static int write_metric_data(const char* filename, fty_proto_t* metric)
     return 0;
 }
 
-int fty::shm::write_metric(fty_proto_t* metric)
+int fty_shm_write_metric_proto(fty_proto_t* metric)
 {
     char filename[PATH_MAX];
 
-    if (prepare_filename(filename, fty_proto_name(metric), strlen(fty_proto_name(metric)), fty_proto_type(metric),
-            strlen(fty_proto_type(metric)), FTY_SHM_METRIC_TYPE) < 0)
+    if (prepare_filename(filename, sizeof filename, fty_proto_name(metric), fty_proto_type(metric), FTY_SHM_METRIC_TYPE) < 0)
         return -1;
+
     return write_metric_data(filename, metric);
 }
 
-int fty_shm_write_metric_proto(fty_proto_t* metric)
+int fty::shm::write_metric(fty_proto_t* metric)
 {
     char filename[PATH_MAX];
 
-    if (prepare_filename(filename, fty_proto_name(metric), strlen(fty_proto_name(metric)), fty_proto_type(metric),
-            strlen(fty_proto_type(metric)), FTY_SHM_METRIC_TYPE) < 0)
+    if(metric == nullptr)
+        return -1;
+    
+    if (prepare_filename(filename, sizeof filename, fty_proto_name(metric), fty_proto_type(metric), FTY_SHM_METRIC_TYPE) < 0)
         return -1;
+
     return write_metric_data(filename, metric);
 }
 
 int fty::shm::write_metric(
     const std::string& asset, const std::string& metric, const std::string& value, const std::string& unit, int ttl)
 {
-    char filename[PATH_MAX];
+    int result;
+    fty_proto_t * prt = fty_proto_new(FTY_PROTO_METRIC);
 
-    if (prepare_filename(
-            filename, asset.c_str(), asset.length(), metric.c_str(), metric.length(), FTY_SHM_METRIC_TYPE) < 0)
+    if( asset.empty() || metric.empty() || value.empty() || unit.empty() )
```

Comment:
i'm certain that unit can be empty...
*must be checked* with UT on the current release/2.7 lib...